File: src/com/suushiemaniac/cubing/alglib/modify/rotate/Rotation.java
package com.suushiemaniac.cubing.alglib.modify.rotate;

import com.suushiemaniac.cubing.alglib.modify.Modifier;
import com.suushiemaniac.cubing.alglib.move.Move;

public interface Rotation extends Modifier {
    Move rotate(Move origin);
}

File: src/main/kotlin/com/suushiemaniac/cubing/alglib/transform/rotate/Rotation.kt
package com.suushiemaniac.cubing.alglib.transform.rotate

import com.suushiemaniac.cubing.alglib.transform.Transform
import com.suushiemaniac.cubing.alglib.move.Move

interface Rotation : Transform {
    fun rotate(origin: Move): Move

    override fun transform(origin: Move) = rotate(origin)
}
